api/planid/plan_id_test.go

```diff
@@ -0,0 +1,52 @@
+package planid
+
+import (
+	"testing"
+)
+
+func TestGetPlanID(t *testing.T) {
+	tests := []struct {
+		name     string
+		revision string
+		want     string
+	}{
+		{
+			name:     "Valid hash more than 10 characters",
+			revision: "branch1@algo1:12345678901234567890",
+			want:     "plan-branch1-1234567890",
+		},
+		{
+			name:     "Valid hash equal to 10 characters",
+			revision: "branch2@algo2:1234567890",
+			want:     "plan-branch2-1234567890",
+		},
+		{
+			name:     "Valid hash less than 10 characters",
+			revision: "branch3@algo3:123456789",
+			want:     "plan-branch3-123456789",
+		},
+		{
+			name:     "Valid branch and hash",
+			revision: "branch4@algo4:abc123xyz",
+			want:     "plan-branch4-abc123xyz",
+		},
+		{
+			name:     "Valid old format",
+			revision: "main/12345678901234567890",
+			want:     "plan-main-1234567890",
+		},
+		{
+			name:     "Valid old format bucket format",
+			revision: "12345678901234567890",
+			want:     "plan-1234567890",
+		},
+	}
```

Comment:
Should you also have some invalid cases?

Reply:
The invalid case is the last one, where you will get the first ten characters as the suffix after `plan-`.

Reply:
Should it be named different then, as it says it's valid?